CONFIGURATION.md

```diff
@@ -0,0 +1,128 @@
+
+## Limitations
+
+The current implementation has several important limitations which it is important to be aware of. First of all, the current implementation is an NTP client only, and does not support acting as an NTP server.
+
+Second, there are currently a few limitations around configuring connections to other NTP nodes:
+ - Only Client-Server connections are supported
+ - DNS lookup is currently only done at startup. Changes in the IP address 
+ - There is no support for NTP pools yet. Multiple 
+ - Changes in network interfaces are not picked up dynamically and will require a reboot of the daemon.
+
+Finally, please be aware that this is very early stage software, and is not yet ready for production software. This software SHOULD NOT be used on any system where you cannot handling experiencing issues, either reliability or security related.
+
+## Building
+
+Currently, NTPD-rs only support linux based operating systems. Our current testing only targets linux kernels after version 5.0.0, older kernels may work but are not guaranteed.
+
+NTPD-rs is written in rust, and requires cargo 1.61.0 at a minimum to be built. We strongly recommend using [rustup](https://rustup.rs) to install rust/cargo, as the version provided by system package managers tend to be out of date.
+
+To build NTPD-rs run
+```sh
+cargo build --release
+```
+This produces a binary `ntp-daemon` in the `target/release` folder, which is the main ntp daemon. The daemon requires elevated permissions in order to change the system clock. It can be tested against a server in the [ntp pool](https://ntppool.org) (please ensure no other ntp daemons are running)
+```sh
+sudo ./target/release/ntp-daemon -p pool.ntp.org
+```
+After a few minutes you should start to see messages indicating the offset of your machine from the server.
+
+## Configuration
+
+The ntp-daemon binary can be configured through two channels: via command line options and via a configuration file. The command line options are primarily intended to tell ntp-daemon where to find its configuration file, and to override the most important settings when debugging problems. The configuration file is the preferred method of configuring ntp-daemon, and allows changing of settings not availble through the command line.
+
+### Command line options
+
+The following command line options are available. When an option is not provided, the indicated default is used.
+
+| Option | Default | Description |
+| --- | --- | --- |
+| `-c <FILE>`, `--config <FILE>` | First existing of ./ntp.toml, /etc/ntp.toml | Which configuration file to use. When provided, the fallback locations are not used. |
+| `-l <LEVEL>`, `--log-filter <LEVEL>` | From configuration file | Set the ammount of information logged, overrides the configuration file. Avaible levels: trace, debug, info, warn, error |
```

Comment:
"Avaible" -> "Available". Instead of explaining this here, consider just referencing the `log_filter` config option instead in order to deduplicate explanations.

You're ending the description with a "." here, but not in the preceding row. Might be nice to be consistent here.

Reply:
fixed